Example mod/Mod.cs

```diff
@@ -1,123 +1,95 @@
     public static class ExampleMod
     {
-        public static Config Config { get; } = new Config();
+        internal static Config Config { get; } = OptionsPanelHandler.RegisterModOptions<Config>();
 
         [QModPatch]
         public static void Patch()
         {
-            Config.Load();
-            OptionsPanelHandler.RegisterModOptions(new Options());
+            Logger.Log(Logger.Level.Info, "Patched successfully!");
         }
     }
 
+    public enum CustomChoice { One, Two, Three }
+
+    [Menu("SMLHelper Example Mod")]
     public class Config : ConfigFile
     {
-        public int ChoiceIndex { get; internal set; }
-        public KeyCode KeybindKey { get; internal set; }
-        public float SliderValue { get; internal set; }
-        public bool ToggleValue { get; internal set; }
-    }
+        [Label("My index-based choice"), Tooltip("A simple tooltip"), Choice("Choice 1", "Choice 2", "Choice 3"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public int ChoiceIndex;
 
-    public class Options : ModOptions
-    {
-        public Options() : base("SMLHelper Example Mod")
-        {
-            ChoiceChanged += Options_ChoiceChanged;
-            KeybindChanged += Options_KeybindChanged;
-            SliderChanged += Options_SliderChanged;
-            ToggleChanged += Options_ToggleChanged;
+        [Label("My string-based choice"), Choice("Foo", "Bar"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public string ChoiceString = "Foo";
 
-            GameObjectCreated += Options_GameObjectCreated;
-        }
+        [Label("My enum-based choice"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public CustomChoice ChoiceEnum;
 
-        public void Options_ChoiceChanged(object sender, ChoiceChangedEventArgs e)
-        {
-            if (e.Id != "exampleChoice") return;
-            ExampleMod.Config.ChoiceIndex = e.Index;
-            ExampleMod.Config.Save();
-        }
-        public void Options_KeybindChanged(object sender, KeybindChangedEventArgs e)
-        {
-            if (e.Id != "exampleKeybind") return;
-            ExampleMod.Config.KeybindKey = e.Key;
-            ExampleMod.Config.Save();
-        }
-        public void Options_SliderChanged(object sender, SliderChangedEventArgs e)
-        {
-            if (e.Id != "exampleSlider") return;
-            ExampleMod.Config.SliderValue = e.Value;
-            ExampleMod.Config.Save();
-        }
-        public void Options_ToggleChanged(object sender, ToggleChangedEventArgs e)
-        {
-            if (e.Id != "exampleToggle") return;
-            ExampleMod.Config.ToggleValue = e.Value;
-            ExampleMod.Config.Save();
-        }
+        [Label("My customised enum-based choice"), Choice("1", "2", "3"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public CustomChoice ChoiceEnumCustomValues;
+
+        [Label("My keybind"), OnChange(nameof(MyGenericValueChangedEvent))]
+        public KeyCode KeybindKey;
```

Comment:
Also here, there could be an attribute for Keybinds.  
If it doesn't need any parameters to function, it can be as simple as this:
```CS
[Label("My keybind"), Keybind, OnChange(nameof(MyGenericValueChangedEvent))]
```

Reply:
> **NOTE:** There is no `KeybindAttribute` or `ToggleAttribute` as these are merely inferred by the field/property type:
`KeyCode` -> `ModKeybindOption`, `bool` -> `ModToggleOption`

I don't mind adding them if we really want them, it just seems like unnecessary code.

Reply:
Then maybe `Label` is the actual redundant attribute.  
If you need the common functionality, you can make `Slider`, `Toggle`, `KeyBind`, and `Choice` all inherit from a `Label` class or implement an `ILabel` interface.  

Then you'd get something a little cleaner
```CS
[Keybind(My keybind"), OnChange(nameof(MyGenericValueChangedEvent))]
public KeyCode KeybindKey;

[Toggle("My checkbox"), OnChange(nameof(MyCheckboxToggleEvent)), OnChange(nameof(MyGenericValueChangedEvent))]
public bool ToggleValue;
```

Reply:
> Then maybe `Label` is the actual redundant attribute.
> If you need the common functionality, you can make `Slider`, `Toggle`, `KeyBind`, and `Choice` all inherit from a `Label` class or implement an `ILabel` interface.
> 
> Then you'd get something a little cleaner
> 
> ```cs
> [Keybind(My keybind"), OnChange(nameof(MyGenericValueChangedEvent))]
> public KeyCode KeybindKey;
> 
> [Toggle("My checkbox"), OnChange(nameof(MyCheckboxToggleEvent)), OnChange(nameof(MyGenericValueChangedEvent))]
> public bool ToggleValue;
> ```

Good shout, don't know why I didn't think of this :D

Reply:
https://github.com/SubnauticaModding/SMLHelper/pull/173/commits/49f679b6b5125ba4bd0fe15cf0dfea2811557838